### PR TITLE
feat:  Initial Aries EDV provider integration

### PR DESCRIPTION
**Pull request by DRK3**
Now when the SDS URL is provided, then the Aries framework is initialized with an EDV provider instead of a IndexedDB provider, which will be used for all storage operations.

This integration is not yet complete in this commit. The keys used by the EDV provider are newly generated every time agent-sdk is initialized and they are not persisted. Fixing this will require some changes to aries-framework-go.

This commit also removes the old SDS storage code that was added before aries-framework-go had a built-in EDV provider.

Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

File: cmd/agent-js-worker/main.go [hunk 1]
```diff
@@ -496,9 +505,21 @@ func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
 		options = append(options, aries.WithTransportReturnRoute(opts.TransportReturnRoute))
 	}
 
-	store, err := jsindexeddb.NewProvider(opts.DBNamespace)
-	if err != nil {
-		return nil, err
+	var store storage.Provider
+
+	if opts.SDSServerURL == "" {
```
Comment (llorllale) [inline cmd/agent-js-worker/main.go:510]: please use a "storage type" field to switch on.

File: cmd/agent-js-worker/main.go [hunk 2]
```diff
@@ -531,6 +552,35 @@ func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
 	return options, nil
 }
 
+// TODO: Use a persistent key instead of creating a new one every time this starts.
+
+// TODO: Use the KMS from the Aries framework for the EDV provider. Right now this isn't possible since
+//  the KMS in the Aries framework uses the provider passed in via WithStorageProvider for KMS storage.
+//  An EDV server can't use itself for KMS storage as this causes a chicken and egg problem.
+//  aries-framework-go will need to be updated to allow a different provider for just the KMS to resolve this.
```
Comment (llorllale) [inline cmd/agent-js-worker/main.go:560]: @Baha-sk

Comment (llorllale) [reply cmd/agent-js-worker/main.go:560]: I guess we need to separate the concepts of `secrets storage` and `document storage` in the framework.

File: cmd/agent-js-worker/main.go [hunk 3]
```diff
@@ -531,6 +552,35 @@ func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
...
+
+// TODO: Use the Aries Crypto object instantiated in the framework for the EDV provider instead
+//  of creating a new one here. This will require some sort of change to aries-framework-go, since the
+//  Aries Crypto object isn't available until the framework has been initialized.
```
Comment (llorllale) [inline cmd/agent-js-worker/main.go:564]: initialization of the default crypto can be moved up and before storage.

File: cmd/agent-js-worker/main.go (hunk 2 above)
Comment (Baha-sk) [reply cmd/agent-js-worker/main.go:560]: yes, we need a new storage type concept in the framework, probably `secure` and `open` storage types.

I think `document` storage type would confuse the underlying EDV storage, we need types that distinguish between secure storage that's used by EncryptedFormatter and open (insecure) storage used by everything else (the kms storage is an example, which technically uses a secretLock mechanism to secure it, but it's a different kind of `secure` storage)

Comment (llorllale) [reply cmd/agent-js-worker/main.go:560]: FYI indy-sdk just has [non-secrets](https://github.com/hyperledger/indy-sdk/blob/8a1bb3665d9d4927d978ee0b188f90fcee4d1fb7/libindy/src/commands/non_secrets.rs#L12) and everything else

Comment (Baha-sk) [reply cmd/agent-js-worker/main.go:560]: Technically private and symmetric keys are secrets too, but their storage in the kms have its own locking mechanism (not using EDV) in the framework.

The EncryptedFormatter needs the kms to create a proper secret storage. So the agent-sdk has two layers of secret storage (one for EDV used by EncryptedFormatter and one for the kms).

File: cmd/agent-js-worker/main.go (hunk 1 above)
Comment (DRK3) [reply cmd/agent-js-worker/main.go]: Done

File: cmd/agent-js-worker/main.go [hunk 4]
```diff
@@ -531,6 +545,56 @@ func agentOpts(opts *agentStartOpts) ([]aries.Option, error) {
 	return options, nil
 }
 
+func createStore(opts *agentStartOpts) (storage.Provider, error) {
```
Comment (soluchok) [inline cmd/agent-js-worker/main.go:548]: I don't see SDS for `agent-rest` and agent-mobile. See the following lines:

https://github.com/trustbloc/agent-sdk/blob/master/cmd/agent-rest/startcmd/start.go#L232
https://github.com/trustbloc/agent-sdk/blob/master/cmd/agent-mobile/pkg/wrappers/command/aries.go#L122

File: cmd/agent-js-worker/src/agent-rest-client.js [hunk 5]
```diff
@@ -399,18 +399,6 @@ const pkgs = {
             path: "/didclient/create-peer-did",
             method: "POST",
         }
-    },
-    credentialclient: {
```
Comment (llorllale) [inline cmd/agent-js-worker/src/agent-rest-client.js:403]: why is this removed?

File: cmd/agent-js-worker/src/agent-rest-client.js [hunk 6]
```diff
@@ -399,18 +399,6 @@ const pkgs = {
...
-        SaveCredential: {
-            path: "/credentialclient/save-credential",
-            method: "POST",
-        }
-    },
-    presentationclient: {
```
Comment (llorllale) [inline cmd/agent-js-worker/src/agent-rest-client.js:409]: why is this removed?

**Review by llorllale — APPROVED** [review]

File: cmd/agent-js-worker/main.go (hunk 4 above)
Comment (DRK3) [reply cmd/agent-js-worker/main.go:548]: For now I've added TODOs for these tasks, since the SDS integration isn't actually complete yet - it will require some changes to aries first (see the other TODO comments I left in this PR). Once the SDS integration is complete then I will add it to aries-rest and aries-mobile. Note that there are other pieces (e.g. the remote KMS and hub-auth) that will also need to be added to aries-rest and aries-mobile eventually.

**Review by soluchok — APPROVED** [review]
